docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,827 @@
+  But, if these child devices have nested child devices, the parent will have to subscribe to,
+  `tedge/child-1/+/#` and `tedge/child-2/+/#` separately,
+  and when `child-3` is added, dynamically subscribe to `tedge/child-3/+/#` as well.
+
+## Proposal 2: Unified topics for every "thing"
+
+This proposal is built on top of the following assumptions:
+1. Every entity, a child device, descendent child devices or services, must register with thin-edge
+   to get their unique IDs generated, unless these entities can ensure uniqueness across the entire fleet.
+1. No dynamic registration for anything.
+1. Avoiding the entity type(device or service, child or descendent etc) from the topics reduces the payload size as well,
+   as the type of an entity needs to be declared only once and need not be repeated over and over for every message.
+
+The proposal is as follows:
+
+A topic scheme that doesn't differentiate between parent child devices or services and
+does not declare the parent-child relation in the topics either, as follows:
+
+```
+tedge/<entity-id>/...
+```
+
+...where `id` could be the `id` of the main device, child device or a service on any of them.
+The alias `main` can still be used for the main tedge device.
+
+Examples:
+* Main device measurements: `tedge/<tedge-device-id>/telemetry/measurements`
+* Main device service measurements: `tedge/<service-id>/telemetry/measurements`
+* Child device measurements: `tedge/<child-device-id>/telemetry/measurements`
+* Child device service measurements: `tedge/<child-service-id>/telemetry/measurements`
+
+```admonish note
+`tedge/main` could be just used as an alias for `tedge/<tedge-device-id>` for simplicity.
+```
+
+**Pros**
+
+1. Due to the lack of distinction in the topics between parent devices and child devices,
+   it is easier to write code for "the device" irrespective of whether it is deployed as a parent or child.
+1. Minimal payload size as we're only including the `id`s everywhere without repeating any context info.
+
+**Cons**
+
+1. Not easy to differentiate the context(from parent or child) easily from the message.
+1. The `id`s must be unique between all devices and services in a deployment.
+1. Not easy to do subscriptions like: "measurements only from child devices or only from services, excluding parent",
+   without keeping track of all the `id`s of services registered with that child device.
+   Easy wild card subscriptions are not possible at all.
+1. No automatic registration as bootstrapping is mandatory for everything.
+
+## Proposal 3: Namespace extensible topics
+
+This scheme splits the topic levels into two parts:
+* entity/component identification levels
+* data type identification levels
+
+This scheme reserves 4 subtopic levels for entity identification,
+and 2 levels for data type identification such as measurements, alarms, events, commands etc as follows:
+
+```
+te/<name-space-1>/<name-space-2>/<name-space-3>/<name-space-4>/<data-type>/<data-instance-type>
```

Comment:
It would not be so generic. Giving usage *hints* would really help without preventing the user to give another meaning to these *name-space*.

```suggestion
te/<entity-kind>/entity-id>/<component-type>/<component-id>/<data-type>/<data-instance-type>
```

Reply:
Yeah, I tried to keep it generic here and give hints/examples immediately afterwards, so that we don't send them in just one direction right from the start.

Reply:
no longer applicable after the presentation improvements. The design docs mentions a more abstract example, however the reference document keeps it simple for users.